src/app/admin/use-add-product.ts

```diff
@@ -0,0 +1,18 @@
+import { req } from '@/shared/client'
+import { productAttr } from '@/shared/product-types'
+import { Product } from '@prisma/client'
+import { useCallback, useState } from 'react'
+
+export function useAddProduct() {
+  const [product, setProduct] = useState<Product>()
+  const createProduct = useCallback(async (productData: productAttr) => {
+    setProduct(
+      await req('/api/product', {
+        method: 'post',
+        data: productData,
```

Comment:
this still says productData!

Also the data type says `productAttr` when it should be `ProductAttr`